notification-bot/src/index.ts

```diff
@@ -0,0 +1,215 @@
+function initializeStreamClient() {
+  return new StreamClient({
+    url: CLIENT_URL,
+    token: CLIENT_TOKEN,
+    async onReconnect(err, retryCount) {
+      console.log("reconnect", err, retryCount);
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      return { reconnect: true };
+    },
+  });
+}
+
+/**
+ * Initializes the event filter configuration for the Apibara client.
+ * @returns {string} Encoded filter string used to specify which blockchain events to listen to.
+ */
+function initializeFilter() {
+  const governance_contract_address = FieldElement.fromBigInt(BigInt(GOV_CONTRACT_ADDRESS));
+  return Filter.create()
+    .withHeader({ weak: true })
+    .addEvent(new EventFilter().withFromAddress(governance_contract_address))
+    .encode();
+}
+
+/**
+ * Configures the Apibara client with a specific filter and block number.
+ * @param {StreamClient} client The Apibara client instance to configure.
+ * @param {string} filter The encoded filter specifying the events to listen for.
+ * @param {number} block_number The blockchain block number from which to start listening.
+ */
+function configureClient(client: StreamClient, filter: Uint8Array, block_number: number) {
+  client.configure({
+    filter: filter,
+    batchSize: 1,
+    cursor: StarkNetCursor.createWithBlockNumber(block_number),
+  });
+}
+
+/**
+ * Main entry point for the event monitoring script.
+ * Initializes the StreamClient and RpcProvider for connecting to the StarkNet blockchain.
+ * Configures the Apibara client to listen for specific blockchain events and processes them accordingly.
+ */
+async function main() {
+  try {
+    // Apibara streaming
+    const client = initializeStreamClient();
+
+    const provider = new RpcProvider({
+      nodeUrl: constants.NetworkName.SN_SEPOLIA,
+      chainId: constants.StarknetChainId.SN_SEPOLIA,
+    });
+
+    const hashAndBlockNumber = await provider.getBlockLatestAccepted();
+    const block_number = hashAndBlockNumber.block_number;
+
+    //Initialize the filter
+    const filter_test = initializeFilter();
+
+    // Configure the apibara client
+    configureClient(client, filter_test, block_number);
+    await listenToMessages(client);
+
+  } catch (error) {
+    console.error("Initialization failed", error);
+    process.exit(1);
+  }
+}
+
+/**
+ * Listens to blockchain events and handles each submit proposel event as they are received.
+ * @param {StreamClient} client The Apibara client instance from which to listen for messages.
+ */
+async function listenToMessages(client: StreamClient) {
+  for await (const message of client) {
+    if (message.message === "data" && message.data?.data) {
+      for (const data of message.data.data) {
+        const block = v1alpha2.Block.decode(data);
+
+        const { header, events, transactions } = block;
+
+        if (!header || !transactions) {
+          continue;
+        }
+
+        for (const event of events) {
+          console.log(event.event)
```

Comment:
This is mostly a debug print and shouldn't be present in the finished code, maybe behind a envvar/config flag

Reply:
done